tests/optimization/test_buffer_pool.py

```
@@ -0,0 +1,114 @@
+        assert pool.buffer_size == 1024 * 1024
+        assert pool.initial_buffers == 10
+        assert pool.total_buffers == 10
+        assert pool.available_buffers == 10
+        assert pool.in_use_count == 0
+
+    def test_invalid_args_raise(self) -> None:
+        with pytest.raises(ValueError, match="buffer_size must be > 0"):
+            BufferPool(buffer_size=0)
+        with pytest.raises(ValueError, match="initial_buffers must be > 0"):
+            BufferPool(initial_buffers=0)
+        with pytest.raises(ValueError, match="max_buffers .* must be >= initial_buffers"):
```

Ruff (0.15.5)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)